File: codingame/community/executionCircle/executionCircle.pl
use strict; use warnings; use 5.20.1;
select(STDOUT); $| = 1; # DO NOT REMOVE

my $tokens;

chomp($tokens=<STDIN>);
my ($n, $s) = split(/ /,$tokens);
chomp(my $d = <STDIN>);

my $x = 2;
my $answer;

while ($x * 2 < $n) {
    $x = $x * 2;
}

if ($d eq "LEFT") {
    $answer = $s + (2 * ($n - $x));
} else {
    $answer = $s - (2 * ($n - $x));
}

if ($answer < 0) {
    $answer = $answer + $n;
}
if ($answer > $n) {
    $answer = $answer - $n;
}
print "$answer\n";

File: codingame/community/executionCircle/executionCircle.rb
n, s = gets.split(" ").collect {|x| x.to_i}
d = gets.chomp

x = 2
while x*2 < n do
    x*=2
end

if d == "LEFT"
    answer = s + (2 * (n - x))
else
    answer = s - (2 * (n - x))
end

if answer < 0
    answer += n
end

if answer > n
    answer -= n 
end

puts answer